packages/@lwc/ssr-runtime/src/lightning-element.ts

```diff
@@ -91,35 +76,31 @@ export class LightningElement implements PropsAvailableAtConstruction {
         return (this.#classList = new ClassList(this));
     }
 
-    #setAttribute(attrName: string, attrValue: string | null): void {
-        this.#attrs[attrName] = attrValue;
-        mutationTracker.add(this, attrName);
-    }
-
-    setAttribute(attrName: string, attrValue: unknown): void {
-        this.#setAttribute(attrName, String(attrValue));
+    setAttribute(attrName: unknown, attrValue: unknown): void {
+        if (typeof attrName === 'string') {
+            const normalizedValue = String(attrValue);
+            this.#attrs[attrName] = normalizedValue;
+            reflectAttrToProp(this, attrName, normalizedValue);
+            mutationTracker.add(this, attrName);
+        }
     }
 
     getAttribute(attrName: unknown): string | null {
-        if (this.hasAttribute(attrName)) {
-            return this.#attrs[attrName as string];
+        if (typeof attrName === 'string' && hasOwnProperty.call(this.#attrs, attrName)) {
+            return this.#attrs[attrName];
         }
         return null;
     }
 
     hasAttribute(attrName: unknown): boolean {
-        return typeof attrName === 'string' && typeof this.#attrs[attrName] === 'string';
-    }
-
-    removeAttribute(attrName: string): void {
-        if (this.hasAttribute(attrName)) {
-            // Reflected attributes use accessor methods to update their
-            // corresponding properties so we can't simply `delete`. Instead,
-            // we use `null` when we want to remove.
-            this.#setAttribute(attrName, null);
-        } else {
-            // This interprets the removal of a non-existing attribute as an
-            // attribute mutation. We may want to revisit this.
+        return typeof attrName === 'string' && hasOwnProperty.call(this.#attrs, attrName);
```

Comment:
```suggestion
        return attrName in this.#attrs;
```

Reply:
Trying to avoid returning inherited properties